### ISSUE TITLE
Which SSL/TLS library is Cloudflare using?

### ISSUE THREAD

**Issue by HamJin** [cloudflare/sslconfig#79]
I used the patch in sslconfig/patches for openssl-1.0.2, but I can't add Curve25519
So , which SSL/TLS library is Cloudflare using?

**Comment by RoldanLT** [comment]
BoringSSL i think.

**Comment by Injust** [comment]
@railjty I'm not sure about multi certs, but OCSP stapling was never removed in BoringSSL. The OCSP protocol was removed, but not stapling and parsing.

**Comment by railjty** [comment]
Now where can I get the patch for boringssl?

**Comment by ymshenyu** [comment]
i think they are using openssl with equal cipher patch. you can get the patch at [https://github.com/hakasenyang/openssl-patch](url)

**Comment by injust** [comment]
@ymshenyu Incorrect, see https://github.com/cloudflare/sslconfig/issues/78#issuecomment-324434709. Cloudflare is using BoringSSL.

**Comment by ymshenyu** [comment]
ok , but boringssl ocsp stapling also need a patch and i am not recommend to use that patch.

**Comment by railjty** [comment]
@ymshenyu  That page is incorrect

**Comment by railjty** [comment]
Now I 'm using the openssl 1.1.1-pre2 with the patch at https://github.com/kn007/patch

**Comment by ymshenyu** [comment]
ok i use hakasenyang's patch with openssl 1.1.1-pre8. that patch can enable tls 1.3 draft 23 too and equal preference cipher groups like boringssl. that patch is good too.

**Comment by hakasenyang** [comment]
Hello.

My patch is **[here](https://github.com/hakasenyang/openssl-patch/blob/master/openssl-equal-pre8_ciphers.patch)**.
Please, [READ ME](https://github.com/hakasenyang/openssl-patch/blob/master/README.md).

**Comment by railjty** [comment]
Tnanks a lot! It's very useful that add tls1.3 draft 23 back. And would someone add chacha20-poly1305-draft cipher back? Like BoringSSL branch 2987, it has both TLS1.3 Draft 18 and chacha20-poly1305-old(0xcc13 0xcx14 0xcc15)

**Comment by hakasenyang** [comment]
@railjty As I think, I no longer need a draft version of chacha20-poly1305.
If you need it, consider **BoringSSL**.

**Comment by railjty** [comment]
Now BoringSSL deleted it. So only old LibreSSL/BoringSSL and cloudflare's patch can add it

**Comment by ymshenyu** [comment]
@injust may i get cloudflare boringssl patch ?
